addons/sale_timesheet/models/project.py

```diff
@@ -537,7 +537,7 @@ def _get_default_partner_id(self, project, parent):
                 return related_project.sale_line_employee_ids.sale_line_id.order_partner_id[:1]
         return res
 
-    sale_order_id = fields.Many2one(domain="['|', '|', ('partner_id', '=', partner_id), ('partner_id', 'child_of', commercial_partner_id), ('partner_id', 'parent_of', partner_id)]")
+    sale_order_id = fields.Many2one(domain="['|', ('partner_id', '=', partner_id), ('commercial_partner_id', 'parent_of', partner_id)]")
```

Comment:
isn't `('partner_id', 'parent_of', partner_id)` less restrictive than `('commercial_partner_id', 'parent_of', partner_id)`?

if we have eg.

- company A:
-- company B
--- has child partner C
---- which has child partner D

then if the partner_id is the current record is D, I get:

- with `('commercial_partner_id', 'parent_of', partner_id)` => partner D, company B
- with `('partner_id', 'parent_of', partner_id)` => partner D, partner C, company B, company A

also it doesn't seems like we can use `commercial_partner_id` since it's added by sale_subscription which is not a dependency of sale_timesheet:

https://github.com/odoo/enterprise/blob/071d19332b5bb6cdb96dafd64fe463e2ed0eef49/sale_subscription/models/sale_order.py#L105

we could use `('partner_id.commercial_partner_id', 'parent_of', partner_id)` but this doesn't solve the issue that what we wanted was `('partner_id', 'child_of', partner_id.commercial_partner_id)`

Reply:
Based on https://github.com/odoo/odoo/blob/17.0/odoo/addons/base/models/res_partner.py#L428-L434 I think it would be:
- `('commercial_partner_id', 'parent_of', partner_id)` => partner D, partner C, company B
But you're right, it doesn't include company A which it should.

Also, I thought that `('partner_id.commercial_partner_id', 'parent_of', partner_id)` and `('partner_id', 'child_of', partner_id.commercial_partner_id)` were equivalent, as in it only checks whether both partners have  a parent in common. Am I misunderstanding something ?

|                                                    |    |      |    |     |
| ---------------------------------------- | --- | -- | -- | -- |
| task.partner_id                        | A | B | C | D |
| task.commercial_partner_id | A | B | B | B |
| so.partner_id                           | A | B | C | D |
| so.commercial_partner_id    | A | B | B | B |

Reply:
I could have thought that also (and it's what I thought at first).  But testing it doesn't seem to be the case.

With my example and checking step by step in odoo/osv/expression.py:

company A:
-- company B
--- has child partner C
---- which has child partner D
---- which has child partner E
company F

('partner_id.commercial_partner_id', 'parent_of', partner_id)

When we have partner_id that is "partner D" we will:

1) this will search `('partner_id', 'any', [('commercial_partner_id', 'parent_of', <partner D>)])`
2) this will search on res.partner [('commercial_partner_id', 'parent_of', <partner D>)]
3) this will return res.partner that are commercial_partner_id of partner_id and itself => "partner D", "company B"
4) then inserting it in the original query we will have `('partner_id', 'in', [<partner D>, <company B>])])`
=> which gives us just the same order with the partner being the given partner or exactly the parent

While `('partner_id', 'child_of', partner_id.commercial_partner_id)` gives us all the sale order id of a partner that is a children of the commercial partner ID.

But after checking understand this in more detail, it seems that what we want would be possible with:

```py
('partner_id.commercial_partner_id.id', 'parent_of', <partner D>)
```
In this case, we will:

- first search the res.partner parent of <partner D> -> "company A", "company B" "partner D", "company B"
- then search the res.partner that have a commercial_partner_id in these results (all besides company F)
- and search the sale.order with a partner_id matching partner_id besides company F